frontend_tests/node_tests/narrow.js

```diff
@@ -92,124 +92,115 @@ run_test("uris", () => {
     assert.equal(emails, "me@example.com");
 });
 
-run_test("show_empty_narrow_message", () => {
+run_test("show_empty_narrow_message", ({mock_template}) => {
     page_params.stop_words = [];
 
+    mock_template("empty_feed_notice.hbs", false, (data) => {
+        const mock_result = "stub_" + data.narrow_id;
+        return mock_result;
+    });
+
     narrow_state.reset_current_filter();
     hide_all_empty_narrow_messages();
     narrow_banner.show_empty_narrow_message();
-    assert.equal($(".empty_feed_notice").visible(), false);
-    assert.ok($("#empty_narrow_message").visible());
+    assert.equal($(".empty_feed_notice_main").html(), "stub_empty_narrow_message");
 
     // for non-existent or private stream
     set_filter([["stream", "Foo"]]);
     hide_all_empty_narrow_messages();
     narrow_banner.show_empty_narrow_message();
-    assert.ok($("#nonsubbed_private_nonexistent_stream_narrow_message").visible());
+    assert.equal(
+        $(".empty_feed_notice_main").html(),
+        "stub_nonsubbed_private_nonexistent_stream_narrow_message",
```

Comment:
What's the thinking behind doing stubs here, rather than mocking the template less aggressively and actually rendering it?

Reply:
I tested changing this to the `data.html` thing you do later in the file, and I think it's a big improvement in the tests (obviously we need to change the output).

Reply:
> What's the thinking behind doing stubs here, rather than mocking the template less aggressively and actually rendering it?

So the existing tests were checking for just the div's id, so I thought it would suffice to do the same for the handlebar template output as well, But yea it would make the test more robust if we add in the HTML check as well.

Reply:
Sounds good, let's test the actual content.